worth2/selftalk/models.py

```diff
@@ -0,0 +1,199 @@
+        return False
+
+    def __unicode__(self):
+        try:
+            slug = self.pageblock().section.get_parent().slug
+        except AttributeError:
+            slug = 'no section'
+
+        return unicode(self.display_name + '[' + slug + '] id: ' +
+                       unicode(self.pk))
+
+    @staticmethod
+    def add_form():
+        return StatementBlockForm()
+
+    def edit_form(self):
+        return StatementBlockForm(instance=self)
+
+    @staticmethod
+    def create(request):
+        form = StatementBlockForm(request.POST)
+        return form.save()
+
+    @classmethod
+    def create_from_dict(cls, d):
+        return cls.objects.create()
+
+    def edit(self, vals, files):
+        form = StatementBlockForm(data=vals, files=files, instance=self)
+        if form.is_valid():
+            form.save()
+
+
+class StatementBlockForm(forms.ModelForm):
+    class Meta:
+        model = StatementBlock
+
+
+class RefutationBlock(BasePageBlock):
+    """A PageBlock for refuting negative statements.
+
+    A self-talk scenario is a custom pageblock containing any number of
+    statements and their corresponding refutations for participants to
+    fill out.
+
+    This block is preceded by the StatementBlock.
+    """
+
+    display_name = 'Self-Talk Refutation Block'
+    template_file = 'selftalk/scenario_block.html'
+
+    statement_block = models.ForeignKey(StatementBlock)
+
+    @property
+    def subject_name(self):
+        return self.statement_block.subject_name
+
+    @property
+    def is_internal(self):
+        return self.statement_block.is_internal
+
+    def needs_submit(self):
+        return True
+
+    def unlocked(self, user):
+        # TODO create logic here
+        return False
+
+    def __unicode__(self):
+        try:
+            slug = self.pageblock().section.get_parent().slug
+        except AttributeError:
+            slug = 'no section'
+
+        return unicode(self.display_name + '[' + slug + '] id: ' +
+                       unicode(self.pk))
+
+    @staticmethod
+    def add_form():
+        return RefutationBlockForm()
+
+    def edit_form(self):
+        return RefutationBlockForm(instance=self)
+
+    @staticmethod
+    def create(request):
+        form = RefutationBlockForm(request.POST)
+        return form.save()
+
+    @classmethod
+    def create_from_dict(cls, d):
+        return cls.objects.create()
+
+    def edit(self, vals, files):
+        form = RefutationBlockForm(data=vals, files=files, instance=self)
+        if form.is_valid():
+            form.save()
+
+
+class RefutationBlockForm(forms.ModelForm):
+    class Meta:
+        model = RefutationBlock
+
+
+class StatementResponse(models.Model):
+    """User's statements they chose, for any self-talk scenario."""
+
+    class Meta:
+        unique_together = ('statement', 'statement_block', 'user')
+
+    statement = models.ForeignKey(Statement)
+    statement_block = models.ForeignKey(StatementBlock)
```

Comment:
Good point... At the same time, I don't think it would make sense for the user to create a `StatementResponse` with no statement block -- there should always be a block they are making the submission to. Maybe this should be a nullable ForeignKey?